### ISSUE TITLE
Pixelated artifacts on feature maps when finetuning the pre-trained model on custom dataset

### ISSUE THREAD

**Issue by dksakkos** [facebook/C3D#366]
The feature maps of the finetuned model are heavily pixelated and causing the output to be very noisy. Screenshots of random feature maps from the layers res2b_branch2a, res2b_branch2b, res3b_branch2b, res4b_branch2b, res5b_branch2b plus the ground truth mask can be found here in that order: https://imgur.com/a/PeCaF

The prototxt remained unchanged, with the exception of appending some deconvolution layers for getting same sized output. Python layers were used for loading 3 consecutive frames into the model (dimension: batch size x channels x frames x width x height = 25x3x3x240x320). The model was initialised with the weights of "c3d_resnet18_sports1m_r2_iter_2800000".

Any ideas what's causing the issue?

**Comment by dutran** [comment]
please use the c++ feature extraction tool, which is well tested. Not sure about the python one.
